tasks/config.yml

```diff
@@ -41,3 +41,11 @@
 - name: Set fact list with custom configuration file
   ansible.builtin.set_fact:
     managed_files: "{{ managed_files | default([]) }} + [ '{{ consul_configd_path }}/50custom.json' ]"
+
+- name: Validate configuration
```

Comment:
Idea (non-blocking): the copy- and template-module have the `validate`-parameter:

```
- validate
        The validation command to run before copying the updated file into the final destination.
        A temporary file path is used to validate, passed in through '%s' which must be present as in the examples below.
        Also, the command is passed securely so shell features such as expansion and pipes will not work.
        For an example on how to handle more complex validation than what this option provides, see Complex configuration validation <https://docs.ansible.com/ansible/devel/reference_appendices/faq.html>.
        default: null
        type: str
```

Maybe we could use this?

Reply:
@rndmh3ro We [discussed this](https://github.com/ansible-collections/ansible-consul/pull/565#discussion_r1285584458) earlier in the PR, actually. The [argument](https://github.com/ansible-collections/ansible-consul/pull/565#discussion_r1285802667) that @bastienwirtz made was that validating on system startup would also protect against external changes to the config file made after the playbook was run.

However, I'm also starting to wonder if maybe just validating the config file during templating would be sufficient. We might have a case of "perfect being the enemy of the good" here. @bastienwirtz, what do you think?

Reply:
I think @rndmh3ro was talking about the possibility to move the validation command into the `ansible.builtin.copy` which support a validate argument (I didn't know, that's cool). it does not relate to the system startup files. It should work, and would be nicer. I can look into it after we fix the issue with the tests. 

@nre-ableton We can remove the startup level validation if you think it helps, but as I understand it, the issue with the tests is related to the fact we decided to clean some unused init system, not that we added a validation command right?

Reply:
@nre-ableton thanks for linking this, I obviously did not see it (I did not check the resolved conversations).

I understand @bastienwirtz' point of view. I also understand your viewpoint for feature-parity in the different OS's.

How about we keep it in the systemd/init-files and move the `Validate configuration` task into the templating/copy-tasks as a `validate`-parameter (this could be done in another PR so this PR here can move forward).

Reply:
> I obviously did not see it (I did not check the resolved conversations).

@rndmh3ro No worries, it was well-hidden. 🙂 I agree that perhaps we could break this work up into two PRs, at least having one for the templating task would be nice.

Reply:
There you go: https://github.com/ansible-collections/ansible-consul/pull/592
Let me know if that works for you!